core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1,3 +1,21 @@
+{{ define "instance" -}}
+{{- $S3URI := self.Parts.s3.Asset.URL -}}
```

Comment:
Assuming the only reference to `self` is this, I prefer `parts.s3.Asset.URL` rather than `self.Parts.s3.Asset.URL` so that we won't expose too much things to the cloud-config template.

Currently, `self` is only used to access `Parts`, right? Then, we can safely remove `self` in favor of an another func for directly accessing `Parts`(would be named `parts` in the FuncMap)?